bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccBindingConstants.java

```diff
@@ -21,13 +21,16 @@
  * used across the whole binding.
  *
  * @author Florian Hotze - Initial contribution
+ * @author Luca Arnecke - Update to evcc version 0.123.1
  */
 @NonNullByDefault
 public class EvccBindingConstants {
 
     private static final String BINDING_ID = "evcc";
 
     public static final String CHANNEL_GROUP_ID_GENERAL = "general";
+    public static final String CHANNEL_GROUP_ID_LOADPOINT = "loadpoint";
+    public static final String CHANNEL_GROUP_ID_VEHICLE = "vehicle";
```

Comment:
vehicle group is not used in `thing-types.xml`

Reply:
I didn't know how to implement the grouping for vehicles. For the loadpoints there is a unconvential solution with loadpoint0 to loadpoint9:

`
<channel-groups>
  <channel-group id="loadpoint0" typeId="loadpoint"/>
  <channel-group id="loadpoint1" typeId="loadpoint"/>
  <channel-group id="loadpoint2" typeId="loadpoint"/>
</channel-groups>

<channel-group-type id="loadpoint">
  <label>Loadpoint</label>
</channel-group-type>
`

You can't do that for vehicles because they don't have numeric ids.

Reply:
According to #15744 there should be even distinguished between Vehicles and Heatingsystems.

Reply:
I already read the openhab binding documentation for the thing-types.xml, but it seems to me that there is nothing like a section. What I saw was the grouping by typeIds like above. It's my first time working on a binding, correct me if I'm wrong.

I don't have a heater implemented in evcc so for me it's difficult to test funcionality. Even in the demo application (demo.evcc.io) there is a heat pump integrated but the api flag "chargerFeatureHeating" is set to false. Did you experienced also problems like this with the newer versions of evcc?

Reply:
I'm also no expert and didn't find a quick solution.
We could add at least one single `vehicle` group, that contains all vehicle specific channels.
In this case, you need to add a `vehicle` group to `thing-types.xml` and remove in two places the `+ vehicleName` from 
`final String channelGroup = "vehicle" + vehicleName;`

Reply:
I also gave #15744 a try. It has again a bigger impact across the entire binding, as several channels, constants ... must be adapted. I think it makes sense to implement #15744 after this one gets finally merged.

Reply:
> I'm also no expert and didn't find a quick solution.
> 
> ~We could add at least one single `vehicle` group, that contains all vehicle specific channels.~ ~In this case, you need to add a `vehicle` group to `thing-types.xml` and remove in two places the `+ vehicleName` from~ ~`final String channelGroup = "vehicle" + vehicleName;`~
> 
> Update: doesn't work, as this results in non unique channel UIDs.

Thats sad, but yeah, that was pretty my problem. Maybe there is a workaround in the future :)

Reply:
I just had a look at the ThingBuilder, it also does not provide a way of dynamically creating channel groups.

@openhab/add-ons-maintainers Do you know any way to dynamically create channel groups?

Reply:
As workaround you could start with # the vehicle channels, then continue # with General, ...
In this case all the Groups will be separated.

Reply:
That could indeed work, but starting with the vehicles followed by general seems really odd to me.